airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -230,6 +248,10 @@ def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mappi
         Unexpected persistent exceptions are not handled and will cause the sync to fail.
         """
         response: requests.Response = self._session.send(request, **request_kwargs)
+        if not self.raise_on_http_errors:
```

Comment:
shouldn't this be around the line `response.raise_for_status()` ? In its current state it is preventing backoff from happening which is not the intention. The intention is to allow the user to handle non-transient exceptions (e.g skip them)

Reply:
I thought that the idea is to skip each http code error.
To clarify, the goal is to skip all http errors except those defined in `should_retry` logic ?
```python
return response.status_code == 429 or 500 <= response.status_code < 600
```

Reply:
correct